hikari/impl/bot.py

```diff
@@ -341,7 +342,7 @@ def __init__(
 
         # Caching
         cache_settings = cache_settings if cache_settings is not None else config.CacheSettings()
-        self._cache = cache_impl.CacheImpl(self, cache_settings)
+        self._cache = cache or cache_impl.CacheImpl(self, cache_settings)
```

Comment:
How are you meant to pass a separate cache impl to the bot's init if you're going to need the bot instance to initiate most custom cache impls (esp any which extend the standard impl)

Reply:
Oh yeah I forgot about that. Let me change that up on my next commit